### PR TITLE
Swift: create common `ErrorElement` superclass and tests

### PR DESCRIPTION
**Pull request by redsun82**

### CODE REVIEW

**Review by MathiasVP — APPROVED** [review]
LGTM after the merge conflict has been resolved!

**Review by MathiasVP — DISMISSED** [review]

**Review by MathiasVP — APPROVED** [review]

**Review by AlexDenisov — APPROVED** [review]

**Pull request merged**
